zcash_client_sqlite/src/wallet.rs

```diff
@@ -2271,10 +2345,11 @@ pub(crate) fn put_tx_data(
     tx: &Transaction,
     fee: Option<NonNegativeAmount>,
     created_at: Option<time::OffsetDateTime>,
-) -> Result<i64, SqliteClientError> {
+    target_height: Option<BlockHeight>,
+) -> Result<TxRef, SqliteClientError> {
     let mut stmt_upsert_tx_data = conn.prepare_cached(
-        "INSERT INTO transactions (txid, created, expiry_height, raw, fee)
-        VALUES (:txid, :created_at, :expiry_height, :raw, :fee)
+        "INSERT INTO transactions (txid, created, expiry_height, raw, fee, target_height)
+        VALUES (:txid, :created_at, :expiry_height, :raw, :fee, :target_height)
         ON CONFLICT (txid) DO UPDATE
         SET expiry_height = :expiry_height,
             raw = :raw,
```

Comment:
Do we also need `target_height = IFNULL(:target_height, target_height)` in the ON CONFLICT clause?

Reply:
This should be set at the same time as `created_at`; I can't think of a situation where the target height would be set later.